src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -220,6 +231,23 @@ func (a *App) onBootstrapControlPlane(s *state.State, initConfig map[string]stri
 		AllowSelfSignedCA:         true,
 		IncludeMachineAddressSANs: true,
 	})
+
+	certificates.CACert = bootstrapConfig.CACert
+	certificates.CAKey = bootstrapConfig.CAKey
+
+	certificates.FrontProxyCACert = bootstrapConfig.FrontProxyCACert
+	certificates.FrontProxyCAKey = bootstrapConfig.FrontProxyCAKey
+
+	certificates.ServiceAccountKey = bootstrapConfig.ServiceAccountKey
+
+	certificates.APIServerKubeletClientCert = bootstrapConfig.APIServerKubeletClientCert
+	certificates.APIServerKubeletClientKey = bootstrapConfig.APIServerKubeletClientKey
+
+	certificates.APIServerCert = bootstrapConfig.APIServerCert
+	certificates.APIServerKey = bootstrapConfig.APIServerKey
+	certificates.KubeletCert = bootstrapConfig.KubeletCert
+	certificates.KubeletKey = bootstrapConfig.KubeletKey
```

Comment:
i wish i knew a good reason why all this copying is going on.  Moving things from one datatype to another....sheesh

Reply:
let's keep it as-is. flat is easier to iterate with, we can add more structure to the code once we are comfortable that the interfaces are well tested.